Model/Export/CanceledOrders.php

```diff
@@ -0,0 +1,274 @@
+        SearchCriteriaBuilder $searchCriteriaBuilder,
+        FilterBuilder $filterBuilder,
+        SortOrderBuilder $sortOrderBuilder,
+        UrlFinderInterface $urlFinder,
+        StoreManagerInterface $storeManager,
+        OrderRepositoryInterface $orderRepositoryInterface,
+        ShipmentRepositoryInterface $shipmentsService,
+        ProductRepository $productRepository,
+        Product $productHelper,
+        DirectoryList $directoryList,
+        TurnToProductHelper $turnToProductHelper,
+        OrderCollectionFactoryAlias $orderCollectionFactory
+    ) {
+        parent::__construct(
+            $config,
+            $productCollectionFactory,
+            $logger,
+            $dateTimeFactory,
+            $searchCriteriaBuilder,
+            $filterBuilder,
+            $sortOrderBuilder,
+            $urlFinder,
+            $storeManager,
+            $orderRepositoryInterface,
+            $shipmentsService,
+            $productRepository,
+            $productHelper,
+            $directoryList,
+            $turnToProductHelper
+        );
+        $this->orderCollectionFactory = $orderCollectionFactory;
+    }
+
+    /**
+     * @param           $storeId
+     * @param \DateTime $fromDate
+     * @param \DateTime $toDate
+     * @param bool      $forceIncludeAllItems
+     *
+     * @return bool|string|null
+     */
+    public function getCanceledOrdersFeed(
+        $storeId,
+        \DateTime $fromDate,
+        \DateTime $toDate,
+        $forceIncludeAllItems = false
+    ) {
+        $csvData = null;
+        $canceledOrders = $this->getCanceledOrders($storeId, $fromDate, $toDate);
+
+        try {
+            $outputFile = $this->directoryList->getPath(DirectoryList::TMP) . '/' . self::CANCELED_FEED_NAME;
+            $outputHandle = fopen($outputFile, 'w+');
+            fputcsv(
+                $outputHandle,
+                [
+                    'ORDERID',
+                    'SKU'
+                ],
+                "\t"
+            );
+            $this->writeOrdersToFeed($outputHandle, $canceledOrders, $forceIncludeAllItems);
+            rewind($outputHandle);
+            $csvData = stream_get_contents($outputHandle);
+        } catch (\Exception $e) {
+            $this->logger->error(
+                'An error occurred while processing Historical Orders Feed Cron',
+                [
+                    'storeId' => $storeId,
+                    'exception' => $e
+                ]
+            );
+        } finally {
+            if (isset($outputHandle)) {
+                fclose($outputHandle);
+            }
+        }
+
+        return $csvData;
+    }
+
+    /**
+     * CRON handler that sends the last 2 days of orders to TurnTo
+     */
+    public function cronUploadFeed()
+    {
+        foreach ($this->storeManager->getStores() as $store) {
+            if ($this->config->getIsEnabled($store->getCode()) && $this->config->getIsHistoricalOrdersFeedEnabled(
```

Comment:
Because this has to do with the order export, i'm assuming that triggering it when the historical order feed is set is correct. Let me know if that's a incorrect assumption.